homeassistant/components/alert.py

```diff
@@ -0,0 +1,294 @@
+    vol.Optional(CONF_STATE): cv.string,
+    vol.Required(CONF_REPEAT): cv.positive_int,
+    vol.Optional(CONF_BACKOFF): vol.Any(int, float),
+    vol.Optional(CONF_CAN_ACK): cv.boolean,
+    vol.Optional(CONF_SKIP_FIRST): cv.boolean,
+    vol.Required(CONF_NOTIFIERS): cv.ensure_list})
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_ENTITIES): [ALERT_SCHEMA],
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+ALERT_SERVICE_SCHEMA = vol.Schema({
+    vol.Required(ATTR_ENTITY_ID): cv.entity_ids,
+})
+
+
+def is_on(hass, entity_id):
+    """Return if the alert is firing and not acknowledged."""
+    return hass.states.is_state(entity_id, STATE_ON)
+
+
+def turn_on(hass, entity_id):
+    """Reset the alert."""
+    data = {ATTR_ENTITY_ID: entity_id}
+    hass.services.call(DOMAIN, SERVICE_TURN_ON, data)
+
+
+@callback
+def async_turn_on(hass, entity_id):
+    """Async reset the alert."""
+    data = {ATTR_ENTITY_ID: entity_id}
+    hass.async_add_job(
+        hass.services.async_call(DOMAIN, SERVICE_TURN_ON, data))
+
+
+def turn_off(hass, entity_id):
+    """Acknowledge alert."""
+    data = {ATTR_ENTITY_ID: entity_id}
+    hass.services.call(DOMAIN, SERVICE_TURN_OFF, data)
+
+
+@callback
+def async_turn_off(hass, entity_id):
+    """Async acknowledge the alert."""
+    data = {ATTR_ENTITY_ID: entity_id}
+    hass.async_add_job(
+        hass.services.async_call(DOMAIN, SERVICE_TURN_OFF, data))
+
+
+def toggle(hass, entity_id):
+    """Toggle acknowledgement of alert."""
+    data = {ATTR_ENTITY_ID: entity_id}
+    hass.services.call(DOMAIN, SERVICE_TOGGLE, data)
+
+
+def setup(hass, config):
+    """Setup alert component."""
+    run_callback_threadsafe(hass.loop, async_setup(hass, config))
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Setup alert component."""
+    alerts = config.get(DOMAIN)
+    all_alerts = {}
+
+    @asyncio.coroutine
+    def async_handle_alert_service(service_call):
+        """Handle calls to alert services."""
+        alert_ids = service.extract_entity_ids(hass, service_call)
+
+        for alert_id in alert_ids:
+            alert = all_alerts[alert_id]
+            if service_call.service == SERVICE_TURN_ON:
+                yield from alert.async_turn_on()
+            elif service_call.service == SERVICE_TOGGLE:
+                yield from alert.async_toggle()
+            else:
+                yield from alert.async_turn_off()
+
+    for alert in alerts.get(CONF_ENTITIES):
+        entity = Alert(hass, alert[CONF_NAME],
+                       alert[CONF_ENTITY_ID],
+                       alert.get(CONF_STATE, DEFAULT_STATE),
+                       alert[CONF_REPEAT],
+                       alert.get(CONF_SKIP_FIRST, DEFAULT_SKIP_FIRST),
+                       alert[CONF_NOTIFIERS],
+                       alert.get(CONF_CAN_ACK, DEFAULT_CAN_ACK),
+                       alert.get(CONF_BACKOFF, DEFAULT_BACKOFF))
+        all_alerts[entity.entity_id] = entity
```

Comment:
How are you going to avoid that you are not overwriting an entity here?

To avoid these conflicts, for example the group component has the user specify the object id as a key in a dict:

```
some_alert:
  can_ack: false
```